streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SavePointServiceImpl.java

```diff
@@ -171,6 +218,182 @@ public SavePoint getLatest(Long id) {
+    CompletableFutureUtils.runTimeout(
+            savepointFuture,
+            10L,
+            TimeUnit.MINUTES,
+            savepointResponse -> {
+              if (savepointResponse != null && savepointResponse.savePointDir() != null) {
+                String savePointDir = savepointResponse.savePointDir();
+                log.info("Request savepoint successful, savepointDir: {}", savePointDir);
+              }
+            },
+            e -> {
+              log.error("Trigger savepoint for flink job failed.", e);
+              ApplicationLog log = new ApplicationLog();
+              log.setAppId(application.getId());
+              log.setYarnAppId(application.getClusterId());
+              log.setOptionTime(new Date());
+              String exception = Utils.stringifyException(e);
+              log.setException(exception);
+              if (!(e instanceof TimeoutException)) {
+                log.setSuccess(false);
+              }
+              applicationLogService.save(log);
+            })
+        .whenComplete(
+            (t, e) -> {
+              application.setOptionState(OptionState.NONE.getValue());
+              application.setOptionTime(new Date());
+              applicationService.update(application);
+            });
+  }
+
+  private String getFinalSavepointDir(@Nullable String savepointPath, Application application) {
+    String result = savepointPath;
+    if (StringUtils.isEmpty(savepointPath)) {
+      try {
+        result = this.getSavePointPath(application);
+      } catch (Exception e) {
+        throw new ApiAlertException(
+            "Error in getting savepoint path for triggering savepoint for app "
+                + application.getId(),
+            e);
+      }
+    }
+    return result;
+  }
+
+  @NotNull
+  private Map<String, Object> tryGetRestProps(Application application, FlinkCluster cluster) {
+    Map<String, Object> properties = new HashMap<>();
+
+    if (ExecutionMode.isRemoteMode(application.getExecutionModeEnum())) {
+      Utils.required(
+          cluster != null,
+          String.format(
+              "The clusterId=%s cannot be find, maybe the clusterId is wrong or "
+                  + "the cluster has been deleted. Please contact the Admin.",
+              application.getFlinkClusterId()));
+      URI activeAddress = cluster.getRemoteURI();
+      properties.put(RestOptions.ADDRESS.key(), activeAddress.getHost());
+      properties.put(RestOptions.PORT.key(), activeAddress.getPort());
+    }
+    return properties;
+  }
+
+  private String getClusterId(Application application, FlinkCluster cluster) {
+    if (ExecutionMode.isKubernetesMode(application.getExecutionMode())) {
+      return application.getClusterId();
+    } else if (ExecutionMode.isYarnMode(application.getExecutionMode())) {
+      if (ExecutionMode.YARN_SESSION.equals(application.getExecutionModeEnum())) {
+        Utils.required(
```

Comment:
Utils.notNull(cluster, ...)

Reply:
sorry for the lines.
Because I just check the feature is healthy.